odiglet/pkg/instrumentation/plugin.go

```diff
@@ -82,16 +83,25 @@ func (p *plugin) GetPreferredAllocation(ctx context.Context, request *v1beta1.Pr
 func (p *plugin) Allocate(ctx context.Context, request *v1beta1.AllocateRequest) (*v1beta1.AllocateResponse, error) {
 	res := &v1beta1.AllocateResponse{}
 
+	// calculate the enabled signals from the collectors group status.
+	// in any error, just use empty enabled signals.
+	// If the Allocate returns an error, the pod will not be scheduled which we have to avoid no matter what.
+	enabledSignals := make(map[common.ObservabilitySignal]struct{})
+
 	odigosNs := env.GetCurrentNamespace()
 	nodeCollectorGroup, err := p.odigosKubeClient.OdigosV1alpha1().CollectorsGroups(odigosNs).Get(ctx, k8sconsts.OdigosNodeCollectorCollectorGroupName, metav1.GetOptions{})
 	if err != nil {
-		log.Logger.Error(err, "Failed to get node collector group")
-		return nil, err
-	}
-
-	enabledSignals := make(map[common.ObservabilitySignal]struct{})
-	for _, signal := range nodeCollectorGroup.Status.ReceiverSignals {
-		enabledSignals[signal] = struct{}{}
+		// we should have collectors group created for odigos device to trigger.
+		// however if we don't, just log and do not populate the enabled signals.
+		if apierrors.IsNotFound(err) {
+			log.Logger.V(3).Info("pod with odigos device started, but collectors group not created. disabling all signals for this pod", "collectorGroupName", k8sconsts.OdigosNodeCollectorCollectorGroupName)
```

Comment:
I think this should have a V(0) level, no? to make sure we see it

Reply:
changed this to be an error